test/opsem/const_float.ll

```diff
@@ -0,0 +1,76 @@
+; RUN: %seabmc "%s" 2>&1 | %oc %s
+; RUN: %seabmc --horn-bv2-lambdas --log=opsem3 "%s" 2>&1 | %oc %s
+;; attempt to invoke constant floating point values,
+;; should work after seahorn adds support for floats
+; CHECK: ^sat$
+; XFAIL: *
+; ModuleID = 'const.c'
+source_filename = "const.c"
+target datalayout = "e-m:e-p:32:32-f64:32:64-f80:32-n8:16:32-S128"
+target triple = "i386-pc-linux-gnu"
+
+@i = constant float 0x405907AE20000000, align 4
+
+; Function Attrs: nounwind
+define i32 @main() #0 {
+  %1 = alloca i32, align 4
+  %2 = alloca i32, align 4
+  store i32 0, i32* %1, align 4
+  %3 = bitcast i32* %2 to i8*
+  call void @llvm.lifetime.start.p0i8(i64 4, i8* %3) #4
+  %4 = call i32 @nd()
+  store i32 %4, i32* %2, align 4, !tbaa !3
+  %5 = load i32, i32* %2, align 4, !tbaa !3
+  %6 = sitofp i32 %5 to float
+  %7 = fcmp olt float %6, 0x405907AE20000000
+  br i1 %7, label %8, label %9
+
+; <label>:8:                                      ; preds = %0
+  store i32 100, i32* %2, align 4, !tbaa !3
+  br label %9
+
+; <label>:9:                                      ; preds = %8, %0
+  %10 = load i32, i32* %2, align 4, !tbaa !3
+  %11 = sitofp i32 %10 to float
+  %12 = fcmp oge float %11, 0x405907AE20000000
+  br i1 %12, label %14, label %13
+
+; <label>:13:                                     ; preds = %9
+  call void @__VERIFIER_error() #5
```

Comment:
should it be like this instead? :
```
call void @verifier.assume(i1 %12)
br label %verifier.error
```

Reply:
This would at least be meaningful since SeaHorn understands `verifier.assume` as a special function.

Whether this is what expected depends on a test. SeaHorn checks whether there is an execution from beginning to end of program that satisfies all assumptions along the way. It returns `sat` if that is possible and `unsat` if not.

So whether this piece of code is the right one depends on whether your test expects to test `sat` or `unsat` behavior (this will determine the assume you need), and whether there are other execution paths in the code.